internal_transfers/actions/src/database.ts

```diff
@@ -49,8 +52,39 @@ async function insertTokenImbalances(
   });
 }
 
+export function bigIntMapToJSON<T>(originalMap: Map<T, bigint>): object {
+  return Object.fromEntries(
+    new Map(Array.from(originalMap, ([k, v]) => [k, v.toString()]))
+  );
+}
+export function jsonFromSettlementData(datum: SimulationData): object {
+  return {
+    logs: datum.logs,
+    blockNumber: datum.blockNumber,
+    ethDelta: bigIntMapToJSON(datum.ethDelta),
+  };
+}
+
+async function insertSettlementSimulations(
+  db: ConnectionPool,
+  datum: SettlementSimulationData
+) {
+  await settlement_simulations(db).insertOrIgnore({
+    tx_hash: hexToBytea(datum.txHash),
+    complete: jsonFromSettlementData(datum.full),
+    reduced: jsonFromSettlementData(datum.reduced),
+    winning_settlement: datum.winningSettlement,
+  });
```

Comment:
It might be nice to have a concrete type that tells you what is that you are inserting in the database, also to make the changes in the implicit JSON schema explicit. Afaiu however this database isn't read by this project, so it might be unnecessary for the project.

Reply:
For now we just store JSON. This project doesn't even care or use this simulation data (it was just requested as a form of "checkpoint" data within the process). Its literally just intended to be unstructured raw data.